compiler/dyno/lib/framework/ErrorBase.cpp

```diff
@@ -0,0 +1,215 @@
+/*
+ * Copyright 2021-2022 Hewlett Packard Enterprise Development LP
+ * Other additional copyright holders may be indicated within.
+ *
+ * The entirety of this work is licensed under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ *
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#include "chpl/framework/ErrorBase.h"
+#include "chpl/framework/ErrorWriter.h"
+#include "chpl/parsing/parsing-queries.h"
+#include "chpl/framework/query-impl.h"
+#include "chpl/types/all-types.h"
+#include <sstream>
+
+namespace chpl {
+
+std::string ErrorBase::message() const {
+  std::ostringstream oss;
+  ErrorWriter ew(/* context */ nullptr, oss, ErrorWriter::MESSAGE_ONLY,
+                 /* useColor */ false);
+  write(ew);
+  return oss.str();
+}
+
+Location ErrorBase::location(Context* context) const {
+  std::ostringstream oss;
+  ErrorWriter ew(context, oss, ErrorWriter::MESSAGE_ONLY,
+                 /* useColor */ false);
+  write(ew);
+  return ew.lastLocation();
+}
+
+ID ErrorBase::id() const {
+  std::ostringstream oss;
+  ErrorWriter ew(/* context */ nullptr, oss, ErrorWriter::MESSAGE_ONLY,
+                 /* useColor */ false);
+  write(ew);
+  return ew.lastId();
```

Comment:
first ID ?

Reply:
It's now just `id` in the new `CompatibilityWriter` class.